src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -24,20 +24,39 @@ namespace Microsoft.CodeAnalysis.ExtractInterface
 {
     internal abstract class AbstractExtractInterfaceService : ILanguageService
     {
+        private readonly struct SymbolMapping
+        {
+            public SymbolMapping(
+                Dictionary<ISymbol, SyntaxAnnotation> symbolToDeclarationAnnotationMap, 
+                Solution annotatedSolution, 
+                List<DocumentId> documentIds, 
+                SyntaxAnnotation typeNodeAnnotation)
+            {
+                SymbolToDeclarationAnnotationMap = symbolToDeclarationAnnotationMap;
+                AnnotatedSolution = annotatedSolution;
+                DocumentIds = documentIds;
+                TypeNodeAnnotation = typeNodeAnnotation;
+            }
+
+            public Dictionary<ISymbol, SyntaxAnnotation> SymbolToDeclarationAnnotationMap { get; }
+            public Solution AnnotatedSolution { get; }
+            public List<DocumentId> DocumentIds { get; }
+            public SyntaxAnnotation TypeNodeAnnotation { get; }
+        }
+
         internal abstract Task<SyntaxNode> GetTypeDeclarationAsync(
             Document document,
             int position,
             TypeDiscoveryRule typeDiscoveryRule,
             CancellationToken cancellationToken);
 
-        internal abstract Solution GetSolutionWithUpdatedOriginalType(
-            Solution solutionWithFormattedInterfaceDocument,
+        internal abstract Task<Solution> UpdateMembersWithExplicitImplementationsAsync(
+            Solution unformattedSolution,
+            IReadOnlyList<DocumentId> documentId,
             INamedTypeSymbol extractedInterfaceSymbol,
+            INamedTypeSymbol typeToExtractFrom,
             IEnumerable<ISymbol> includedMembers,
             Dictionary<ISymbol, SyntaxAnnotation> symbolToDeclarationAnnotationMap,
-            List<DocumentId> documentIds,
-            SyntaxAnnotation typeNodeAnnotation,
-            DocumentId documentIdWithTypeNode,
             CancellationToken cancellationToken);
```

Comment:
nit: when i see 'internal' i expect that some other type will call this method.  Is that hte case here, or is this only called from this type itself.  if so, make protected.